backend/scrapers/police_coders_groups/localise.py

```diff
@@ -0,0 +1,46 @@
+import json
+import csv
+from shapely.geometry import shape, Point
+
+def iterateGroups(input_path, out_path, fileNm_Wales):
+    output = []
+    
+    polygon = getWelshBoundary(fileNm_Wales)
+
+    with open(input_path, encoding="utf-8") as f:
+        data = csv.DictReader(f)
+
+        for row in data:
+            point = Point([float(row['Lng']),float(row['Lat'])])
+
+            if polygon.contains(point):
+
+                output.append(row)
+            
+    with open(out_path, "w", encoding="utf-8") as f:  
+        
+        writer = csv.writer(f)
+        
+        #Write csv header
+        writer.writerow(row)
+
+        #Write groups as rows
+        for row in output:
+            writer.writerow(row.values())
+    
+    print("Message (iterateGroups): Localised {} groups to Wales".format(len(output)))
```

Comment:
Now I understand - again mixing processing and I/O.. 

you should now reconnect to my previous concern of having an `iterateXX` function with no return? only side effect.. _bad smell_ again, sorry ;)